jenkins-build-pack-api/Jenkinsfile

```diff
@@ -107,207 +108,128 @@ node() {
-					//validateSwaggerFile( service, domain, apiHostName )
-					// import swagger api spec
 					echo "Deploying to staging API Gateway environment.."
-
-					sh "$aws_api_import --region ${configLoader.AWS.REGION} --update ${configLoader.AWS.API.DEV_ID} --deploy dev swagger/swagger.json"
-
-					def endpointUrl = "https://${configLoader.AWS.API.DEV_ID}.execute-api.${configLoader.AWS.REGION}.amazonaws.com/dev/" + domain + "/" + params.service_name
-					updateServiceWithEndpointInfo(params.service_name, domain, g_base_url + '/platform/services', current_environment, endpointUrl)
-
-          			updateApiDocUrl(current_environment)
-
+					sh "aws apigateway put-rest-api --rest-api-id ${configLoader.AWS.API[api_key]} --mode merge --body 'file://swagger/swagger.json'"+" --profile cloud-api"
+                    sh "aws apigateway create-deployment --rest-api-id ${configLoader.AWS.API[api_key]} --stage-name ${current_environment} --profile cloud-api"
+					
+					def endpointUrl = "https://${configLoader.AWS.API[api_key]}.execute-api.${configLoader.AWS.REGION}.amazonaws.com/${current_environment}/" + domain + "/" + params.service_name
+					if(domain != "platform"){
+						createSubscriptionFilters(serviceName, current_environment, configLoader.AWS.REGION,roleId);
+						updateServiceWithEndpointInfo(config, current_environment, endpointUrl)
+					}
+					updateApiDocUrl(current_environment)
 					sh "aws s3 cp swagger/ s3://${configLoader.AWS.S3.API_DOC}/${domain}/${params.service_name}/${current_environment} --recursive "
-
-					svc_status  = "Your service endpoint for production environment: $endpointUrl \n\nView and test your API here: $g_base_url_for_swagger$domain/$service/dev/swagger.json"
+					echo "completed deployment........."					
+					sendEvent(service, branch, runtime, "PRODUCTION", configLoader.AWS.REGION, domain, configLoader.AWS.ROLEID, Event_Name.DEPLOY_TO_AWS, Event_Status.COMPLETED, "")
+					def svc_status  = "Your service endpoint for ${current_environment} environment: ${endpointUrl} \n\nView and test your API here: ${g_base_url_for_swagger}${domain}/${service}/${current_environment}/swagger.json"
+					send_status_email(config, 'COMPLETED',svc_status)	
 				}
-
-				// @TODO : Resetting the credentials needs to be introduced. (Re-use of same profile on Async calls needs to be considered)
-				// resetCredentials()
-
-				send_status_email(created_by, 'COMPLETED', params.service_name, svc_status)
-				updateServiceInDB(params.service_name, domain, g_base_url + '/platform/services', "active")
-				// @TODO : Send Event feature coming soon
-				sendEvent(service, branch, runtime, "DEVELOPMENT", configLoader.AWS.REGION, domain, configLoader.AWS.ROLEID, Event_Name.DEPLOY_TO_AWS, Event_Status.COMPLETED, "")
-			  }
-			} //Stage to deploy ends here
-
-		}//if branch name ends here
+				if(domain && domain == "platform"){						
```

Comment:
Do we need this if condition?

Reply:
Yes Surya. Because we are setting log stream permission for only "cloud-log-streamer".